PowerEditor/src/WinControls/DockingCont.cpp

```diff
@@ -0,0 +1,1462 @@
+			NMHDR nmhdr;
+			nmhdr.code		= DMN_SWITCHOFF;
+			nmhdr.hwndFrom	= _hSelf;
+			nmhdr.idFrom	= 0;
+			::SendMessage(((tTbData*)tcItem.lParam)->hClient, WM_NOTIFY, nmhdr.idFrom, reinterpret_cast<LPARAM>(&nmhdr));
+		}
+
+		// resize tab item
+
+		// get at first largest item ...
+		HDC		hDc	= ::GetDC(_hContTab);
+		SelectObject(hDc, _hFont);
+
+		for (int iItem = 0; iItem < iItemCnt; ++iItem)
+		{
+			const TCHAR *pszTabTxt = NULL;
+
+			::SendMessage(_hContTab, TCM_GETITEM, iItem, reinterpret_cast<LPARAM>(&tcItem));
+			if (!tcItem.lParam)
+				continue;
+			pszTabTxt = reinterpret_cast<tTbData*>(tcItem.lParam)->pszName;
+
+			// get current font width
+			GetTextExtentPoint32(hDc, pszTabTxt, lstrlen(pszTabTxt), &size);
+
+			if (maxWidth < size.cx) 
+			{
+				maxWidth	= size.cx;
+				pszMaxTxt	= pszTabTxt;
+			}
+		}
+		::ReleaseDC(_hSelf, hDc);
+
+		tcItem.mask	= TCIF_TEXT;
+
+		for (int iItem = 0; iItem < iItemCnt; ++iItem)
+		{
+			generic_string szText;
+			if (iItem == iTab && pszMaxTxt)
+			{
+				// fake here an icon before text ...
+				szText = TEXT("    ");
+				szText += pszMaxTxt;
+			}
+			tcItem.pszText = (TCHAR *)szText.c_str();
+			::SendMessage(_hContTab, TCM_SETITEM, iItem, reinterpret_cast<LPARAM>(&tcItem));
+		}
+
+		// selects the pressed tab and store previous tab
+		::SendMessage(_hContTab, TCM_SETCURSEL, iTab, 0);
+		_prevItem = iTab;
+
+		// update caption text
+		updateCaption();
+
+		onSize();
+	}
+}
+
+bool DockingCont::updateCaption()
+{
+	if (!_hContTab)
+		return false;
+
+	TCITEM			tcItem	= {0};
+	int				iItem	= getActiveTb();
+
+	if (iItem < 0)
+		return false;
+
+	// get data of new active dialog
+	tcItem.mask		= TCIF_PARAM;
+	::SendMessage(_hContTab, TCM_GETITEM, iItem, reinterpret_cast<LPARAM>(&tcItem));
+
+	if (!tcItem.lParam) return false;
+
+	// update caption text
+	_pszCaption = ((tTbData*)tcItem.lParam)->pszName;
+
+	// test if additional information are available
+	if ((((tTbData*)tcItem.lParam)->uMask & DWS_ADDINFO) && 
+		(lstrlen(((tTbData*)tcItem.lParam)->pszAddInfo) != 0))
+	{
+		_pszCaption += TEXT(" - ");
+		_pszCaption += ((tTbData*)tcItem.lParam)->pszAddInfo; 
+	}
+
+	if (_isFloating == true)
+	{
+		::SetWindowText(_hSelf, _pszCaption.c_str());
+	}
+	else
+	{
+		::SetWindowText(_hCaption, _pszCaption.c_str());
+	}
+	return true;
+}
+
+void DockingCont::focusClient()
+{
+	TCITEM		tcItem	= {0};
+	int			iItem	= getActiveTb();	
+
+	if (iItem != -1)
+	{
+		// get data of new active dialog
+		tcItem.mask		= TCIF_PARAM;
+		::SendMessage(_hContTab, TCM_GETITEM, iItem, reinterpret_cast<LPARAM>(&tcItem));
+
+		// set focus
+		if (!tcItem.lParam)
+			return;
+
+		tTbData *tbData = (tTbData *)tcItem.lParam;
+		if (tbData->pszAddInfo && lstrcmp(tbData->pszAddInfo, DM_NOFOCUSWHILECLICKINGCAPTION) == 0)
+			return;
+		
+		::SetFocus(tbData->hClient);
+	}
+}
+
+LPARAM DockingCont::NotifyParent(UINT message)
+{
+	return ::SendMessage(_hParent, message, 0, reinterpret_cast<LPARAM>(this));
+}
+

```

Comment:
"The file added whereas it does exist" is not my fault. It is preexisting condition.  My small changes only make the finder tab name visible, when it is not selected. 
I believe the changes are beneficial, but I am open to remove them.